mlir/lib/Dialect/Tosa/IR/TosaOps.cpp

```diff
@@ -1054,6 +1055,46 @@ LogicalResult tosa::TransposeOp::inferReturnTypeComponents(
   return success();
 }
 
+LogicalResult tosa::TransposeOp::verify() {
+  TensorType inputType = getInput1().getType();
+  TensorType permType = getPerms().getType();
+  TensorType outputType = getOutput().getType();
+
+  if (permType.hasRank() && permType.getRank() != 1)
```

Comment:
The [TOSA spec](https://www.mlplatform.org/tosa/tosa_spec.html#_transpose) does not allow the non-one rank. It makes sense.